client/app/lib/components/extensions/conditions/conditions/AssessmentCondition.tsx

```diff
@@ -124,6 +126,22 @@ const AssessmentConditionForm = (
                   variant="filled"
                 />
               )}
+              renderOption={(optionProps, option): JSX.Element => {
+                const assessment = assessments[parseInt(option, 10)];
+
+                return (
+                  <Box
+                    component="li"
+                    {...optionProps}
+                    key={option}
+                    className={`${optionProps.className} justify-between space-x-4`}
+                  >
+                    <Typography>{assessment}</Typography>
+
+                    <Chip label={`id: ${option}`} size="small" />
```

Comment:
Do we really want to put the id as the chip? It may be a way for user to differentiate 2 different items of the same name, but they may not even know what ID refers to in the first place.. As a user, I feel it adds additional information (ie the ID which is supposedly only known to developers or someone handling DB) to process when I am just simply trying to choose an item. Also, in most cases, there shouldnt be duplicate items of the same name.

Reply:
>Also, in most cases, there shouldnt be duplicate items of the same name.

Not true. The reason why this issue was spotted in the first place is because there are such cases. When @cysjonathan raised the problem of the `Autocomplete` not filtering the options correctly, this is the issue he was observing. It is often that instructors duplicate assessments as back-ups. In fact, duplication will duplicate assessments using the same name _by default_.

>the ID which is supposedly only known to developers or someone handling DB

I agree that ID is a part of our internal implementation details. However, there is a need for us to show the difference between two options of the same title. Even if the "duplicate options" problem in React is solved by attaching `id`s as `key`s, the user can still get confused if they want to choose one out of the two assessments/surveys of the same name.

If you have another idea of what distinguishing information we can use, other than ID, I'm up for it.

Reply:
> Not true. The reason why this issue was spotted in the first place is because there are such cases. When @cysjonathan raised the problem of the Autocomplete not filtering the options correctly, this is the issue he was observing. It is often that instructors duplicate assessments as back-ups. In fact, duplication will duplicate assessments using the same name by default.

Sure, but wont the number of unique items will be “much” more than duplicated items? They do duplicate assessments but not for all right?

I see a few ways for this:
- Add the chip (w or w/o id) only for items with duplicated names and highlight it has duplicates
- Add hyperlinks to the duplicated items
- A combination of both

What do you think?

By only showing the additional info for duplicated items, it wont overwhelm users with the id for other items that are unique.

@cysjonathan any additional thought?

Reply:
>Add the chip (w or w/o id) only for items with duplicated names and highlight it has duplicates

Highlighting it is fine, but there's no use in highlighting it without any distinguishing information. They can already see that there are 2 items of the same name. The problem is that, sometimes, users won't even realise that they have duplicate items, until they search, and then they see it, then two things would happen:
- This must be a buggy software.
- Oh yes, I have two of the same name. But I want *that* one. Which is which?



>Add hyperlinks to the duplicated items

This is okay, but possibly at the expense of context overload. Consider the user flow:
1. I see two assessments of the same name.
2. I click on the hyperlink to see what's the deal with each one of them.
3. After going through both tabs, I found it. *This* is the one.
4. I go back, but which is which again?

Probably the user will just remember "it's the first one". But that is also contingent on the order of the tab maps to the order of the options shown, i.e., (tab1, tab2) -> option1, option2. But the moment they fuzzle the keyword, and the "filtering" is done again, there is no guarantee of such order.


>By only showing the additional info for duplicated items, it wont overwhelm users with the id for other items that are unique.

I did consider only showing the ID chips for duplicate items, but I did not end up doing it because, it might make our code more confusing for very little return (as what @cysjonathan always considers). Plus the algorithm for searching duplicates isn't exactly fast (compared to, well, no search lol).

---
Anyway, these are just my two cents, no objections for any.

Reply:
What about highlighting the duplicates + showing the creation date for only the duplicated assessments? At least the creation date shows more information rather than the ID. One could argue that the item with larger ID means it's created later but not everyone may know this concept.

Reply:
Actually the idea of adding hyperlink is so that users may also change the title if they encounter duplicates. Also, if lets say we decide to change the id to uuid in the future, the text wont fit in the chip anymore

Reply:
As discussed, we will go with adding a hyperlink that allows them to view the details for each assessment and survey in a new tab.